src/html-pipe.js

```diff
@@ -82,13 +82,21 @@ export async function htmlPipe(state, req) {
     }
 
     // ...and apply the folder mapping
-    await folderMapping(state);
+    state.timer?.update('content-fetch');
+    let contentPromise = await fetchContent(state, req, res);
+    // ...but only if the current resource doesn't exist
+    if (res.status === 404) {
+      await folderMapping(state);
+      if (state.info.unmappedPath) {
```

Comment:
`state.mapped` doesn't seem to be set for the code bus, nor for the json pipe, so ended up leveraging the only property that is consistently set.